sklearn/utils/estimator_checks.py

```diff
@@ -488,18 +496,24 @@ def parametrize_with_checks(estimators, strict_mode=True):
                "Please pass an instance instead.")
         raise TypeError(msg)
 
-    def checks_generator():
+    if checks_generator is None:
+        checks_generator = _yield_all_checks
+
+    def _checks_generator():
```

Comment:
Nit but we don't need the newly-added leading underscore here since there's no notion of private/public (it might simplify the diff also)